internal/common/common.go

```diff
@@ -190,3 +193,51 @@ func IsSliceNonEmpty(arg interface{}) bool {
 	}
 	return res
 }
+
+func getUnifiedNTPSources(db *gorm.DB, clusterID strfmt.UUID) (string, error) {
+	var sources []string
+	err := db.Raw("select distinct(additional_ntp_sources) as sources from infra_envs where id in (select distinct(infra_env_id) from hosts where cluster_id = ?)", clusterID.String()).Pluck("sources", &sources).Error
+	if err != nil {
+		return "", err
+	}
+	combinedSources := make(map[string]bool)
+	for _, singleSource := range sources {
+		if singleSource != "" {
+			for _, s := range strings.Split(singleSource, ",") {
+				combinedSources[s] = true
+			}
+		}
+	}
+	var ret []string
+	for k := range combinedSources {
+		ret = append(ret, k)
+	}
+	return strings.Join(ret, ","), nil
+}
+
+func GetClusterNTPSources(db *gorm.DB, clusterID strfmt.UUID) (string, error) {
+	cluster, err := GetClusterFromDB(db, clusterID, SkipEagerLoading)
+	if err != nil {
+		return "", err
+	}
+	if cluster.AdditionalNtpSource != "" {
+		return cluster.AdditionalNtpSource, nil
+	} else {
+		return getUnifiedNTPSources(db, clusterID)
+	}
+}
+
+func GetHostNTPSources(db *gorm.DB, host *models.Host) (string, error) {
+	var (
+		infraEnv *InfraEnv
+		err      error
+	)
+	if host.ClusterID != nil && host.ClusterID.String() != "" {
```

Comment:
you can just check if host.ClusterID != nil. In case it is an empty string, and error will be returned from GetClusterNTPSources, which is ok

Reply:
OK